src/app/qgsapplayertreeviewmenuprovider.cpp

```diff
@@ -272,6 +273,34 @@ QMenu *QgsAppLayerTreeViewMenuProvider::createContextMenu()
         menu->addAction( a );
       }
 
+      // actions on the selection
+      if ( vlayer && vlayer->selectedFeatureCount() > 0 )
+      {
+        int selectionCount = vlayer->selectedFeatureCount();
+        QgsMapLayerAction::Target target;
+        if ( selectionCount == 1 )
+          target = QgsMapLayerAction::Target::SingleFeature;
+        else
+          target = QgsMapLayerAction::Target::MultipleFeatures;
+
+        const QList<QgsMapLayerAction *> constRegisteredActions = QgsGui::mapLayerActionRegistry()->mapLayerActions( vlayer, target );
+        if ( !constRegisteredActions.isEmpty() )
+        {
+          QMenu *actionMenu = menu->addMenu( tr( "Actions on selection (%1)" ).arg( selectionCount ) );
+          for ( QgsMapLayerAction *action : constRegisteredActions )
+          {
+            if ( target == QgsMapLayerAction::Target::SingleFeature )
+            {
+              actionMenu->addAction( action->text(), action, [ = ]() {action->triggerForFeature( vlayer,  new QgsFeature( vlayer->selectedFeatures().at( 0 ) ) );} );
```

Comment:
I believe that this is leaking.
How can I handle this?

Reply:
Change the api -- it shouldn't be a pointer here

Reply:
@nyalldawson done, works for you?